### PR TITLE
chore: polkadot-v1.0.0

### PR DESCRIPTION
**Pull request by Ad96el**
## fixes https://github.com/KILTprotocol/ticket/issues/2800

The changes can be found [here](https://github.com/paritytech/polkadot/releases/tag/v1.0.0).

The Genesis build is now compatible with no_std, so I have updated the serde imports.

I have checked the CLI and the XCM config against the parachain node template.

## Checklist:

- [x] I have verified that the code works
  - [x] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: nodes/parachain/src/command.rs [hunk 1]
```diff
@@ -93,6 +94,14 @@ fn load_spec(id: &str) -> std::result::Result<Box<dyn sc_service::ChainSpec>, St
 	}
 }
 
+fn native_runtime_version(is_spiritnet: bool) -> &'static RuntimeVersion {
```
Comment (Ad96el) [inline nodes/parachain/src/command.rs:97]: Not sure about this. Clippy was complaining.

File: support/Cargo.toml [hunk 2]
```diff
@@ -27,7 +27,7 @@ sp-runtime.workspace = true
 sp-std.workspace = true
 
 # std dependencies
-serde = {workspace = true, optional = true}
+serde = {workspace = true, default-features= false, features=["derive"]}
```
Comment (ntn-x2) [inline support/Cargo.toml:30]: `default-features` is already specified in the root TOML, so you can skip that here.

File: runtimes/spiritnet/src/lib.rs [hunk 3]
```diff
@@ -123,16 +123,14 @@ impl frame_system::Config for Runtime {
 	/// The lookup mechanism to get account ID from whatever is passed in
 	/// dispatchers.
 	type Lookup = AccountIdLookup<AccountId, ()>;
-	/// The index type for storing how many extrinsics an account has signed.
-	type Index = Index;
-	/// The index type for blocks.
-	type BlockNumber = BlockNumber;
+	/// The nonce type for storing how many extrinsics an account has signed.
+	type Nonce = runtime_common::Nonce;
```
Comment (ntn-x2) [inline runtimes/spiritnet/src/lib.rs:127]: Can't you bring `use runtime_common::Nonce`?

File: runtimes/peregrine/src/lib.rs [hunk 4]
```diff
@@ -122,16 +123,14 @@ impl frame_system::Config for Runtime {
 	/// The lookup mechanism to get account ID from whatever is passed in
 	/// dispatchers.
 	type Lookup = AccountIdLookup<AccountId, ()>;
-	/// The index type for storing how many extrinsics an account has signed.
-	type Index = Index;
-	/// The index type for blocks.
-	type BlockNumber = BlockNumber;
+	/// The nonce type for storing how many extrinsics an account has signed.
+	type Nonce = runtime_common::Nonce;
```
Comment (ntn-x2) [inline runtimes/peregrine/src/lib.rs:127]: Same here.

File: runtimes/peregrine/src/lib.rs [hunk 5]
```diff
@@ -698,7 +698,7 @@ impl parachain_staking::Config for Runtime {
 	type NetworkRewardBeneficiary = runtime_common::SendDustAndFeesToTreasury<Runtime>;
 	type WeightInfo = weights::parachain_staking::WeightInfo<Runtime>;
 
-	const BLOCKS_PER_YEAR: Self::BlockNumber = constants::BLOCKS_PER_YEAR;
+	const BLOCKS_PER_YEAR: BlockNumberFor<Self> = constants::BLOCKS_PER_YEAR;
```
Comment (ntn-x2) [inline runtimes/peregrine/src/lib.rs:701]: Why is this different than what's used for Spiritnet?

File: runtimes/kestrel/src/lib.rs [hunk 6]
```diff
@@ -156,16 +156,14 @@ impl frame_system::Config for Runtime {
...
 	/// The type for hashing blocks and tries.
 	type Hash = Hash;
 	/// The hashing algorithm used.
 	type Hashing = BlakeTwo256;
-	/// The header type.
-	type Header = generic::Header<BlockNumber, BlakeTwo256>;
+	/// The block type as expected in this runtime
+	type Block = Block;
+	/// The nonce type for storing how many extrinsics an account has signed.
+	type Nonce = runtime_common::Nonce;
```
Comment (ntn-x2) [inline runtimes/kestrel/src/lib.rs:166]: Same here.

File: pallets/attestation/src/mock.rs [hunk 7]
```diff
@@ -217,13 +214,12 @@ pub(crate) mod runtime {
...
 		type RuntimeCall = RuntimeCall;
-		type Index = u64;
-		type BlockNumber = u64;
 		type Hash = Hash;
 		type Hashing = BlakeTwo256;
 		type AccountId = AccountId;
 		type Lookup = IdentityLookup<Self::AccountId>;
-		type Header = Header;
+		type Block = Block;
+		type Nonce = u32;
```
Comment (ntn-x2) [inline pallets/attestation/src/mock.rs:222]: This was `u64`, right?

File: pallets/ctype/src/mock.rs [hunk 8]
```diff
@@ -81,13 +77,13 @@ pub mod runtime {
 	impl frame_system::Config for Test {
 		type RuntimeOrigin = RuntimeOrigin;
 		type RuntimeCall = RuntimeCall;
-		type Index = u64;
-		type BlockNumber = u64;
+		type Block = Block;
+		type Nonce = u32;
```
Comment (ntn-x2) [inline pallets/ctype/src/mock.rs:81]: Also here. I would keep them `u64` as they were before, if there's no strong reason to change them. We could do so in a different PR.

File: pallets/parachain-staking/src/inflation.rs [hunk 9]
```diff
@@ -20,13 +20,10 @@
 use crate::{pallet::Config, types::BalanceOf};
 use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
-use sp_runtime::{traits::Saturating, Perquintill, RuntimeDebug};
-
-#[cfg(feature = "std")]
 use serde::{Deserialize, Serialize};
+use sp_runtime::{traits::Saturating, Perquintill, RuntimeDebug};
 
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
```
Comment (weichweich) [inline pallets/parachain-staking/src/inflation.rs:28]: why do we remove this here?

**Review by weichweich — APPROVED** [review]
Looks good. Just two things:

- Rather use the u64 nonce so that the tests are closer to our runtime
- Why do we also build serde in non STD environments?

**Comment by weichweich** [comment]
Also run `cargo update`

File: support/Cargo.toml (hunk 2 above)
Comment (Ad96el) [reply support/Cargo.toml]: True - thanks :raised_hands: [4a5345a](https://github.com/KILTprotocol/kilt-node/pull/559/commits/4a5345af379dfbd0179ab78db4eb6fd87a49deb8)

File: runtimes/spiritnet/src/lib.rs (hunk 3 above)
Comment (Ad96el) [reply runtimes/spiritnet/src/lib.rs]: I prefer the runtime_common::Nonce declaration because it makes it clearer where the value is defined, and the code remains clean. But updated in: [94b4a70](https://github.com/KILTprotocol/kilt-node/pull/559/commits/94b4a70489d1a4fa1503e9729089c2036be4ce4d)

File: runtimes/peregrine/src/lib.rs (hunk 4 above)
Comment (Ad96el) [reply runtimes/peregrine/src/lib.rs]: updated in: [94b4a70](https://github.com/KILTprotocol/kilt-node/pull/559/commits/94b4a70489d1a4fa1503e9729089c2036be4ce4d)

File: runtimes/peregrine/src/lib.rs (hunk 5 above)
Comment (Ad96el) [reply runtimes/peregrine/src/lib.rs:701]: Good catch: [c4120f6](https://github.com/KILTprotocol/kilt-node/pull/559/commits/c4120f6f440169d5f3b063613ce84b1240a47f31)

File: runtimes/kestrel/src/lib.rs (hunk 6 above)
Comment (Ad96el) [reply runtimes/kestrel/src/lib.rs]: updated in: [94b4a70](https://github.com/KILTprotocol/kilt-node/pull/559/commits/94b4a70489d1a4fa1503e9729089c2036be4ce4d)

File: pallets/attestation/src/mock.rs (hunk 7 above)
Comment (Ad96el) [reply pallets/attestation/src/mock.rs]: I can not remember why I changed the type… So I reverted it: [74a1a67](https://github.com/KILTprotocol/kilt-node/pull/559/commits/74a1a674b19b7e9ebe95d1832d8ef41fbeb3b8e1)

File: pallets/ctype/src/mock.rs (hunk 8 above)
Comment (Ad96el) [reply pallets/ctype/src/mock.rs]: [74a1a67](https://github.com/KILTprotocol/kilt-node/pull/559/commits/74a1a674b19b7e9ebe95d1832d8ef41fbeb3b8e1)

File: pallets/parachain-staking/src/inflation.rs (hunk 9 above)
Comment (Ad96el) [reply pallets/parachain-staking/src/inflation.rs:28]: Serialization and deserialization are now required in the no_std environment. The genesis_config macro has been updated, as seen here: https://github.com/paritytech/substrate/pull/14306/files#diff-f6aee1a6b4a03b3f4366510598ac506d66658b747049868c583dda3863c484ad

**Review by ntn-x2 — APPROVED** [review]
Fix merge conflicts and then LGTM! 😃
